cmake/wasi-sdk-sysroot.cmake

```diff
@@ -336,6 +336,19 @@ file(GENERATE OUTPUT ${version_file_tmp} CONTENT ${version_dump})
 add_custom_target(version-file DEPENDS ${version_file_tmp})
 add_dependencies(build version-file)
 
+# Install a `version.h` script in each sysroot's include directory.
+execute_process(
+  COMMAND ${PYTHON} ${version_script} header
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+  OUTPUT_VARIABLE version_header)
+foreach(target IN LISTS WASI_SDK_TARGETS)
+  message(STATUS "Creating version.h in ${version_dir}")
+  set(version_header_path ${wasi_sysroot}/include/${target}/wasi/version.h)
+  file(GENERATE OUTPUT ${version_header_path} CONTENT ${version_header})
+  add_custom_target(version-header-${target} DEPENDS ${version_header_path})
+  add_dependencies(build version-header-${target})
+endforeach()
```

Comment:
A more traditional cmake way of doing this would be to create a template file called `version.h` that looks like this:

```
#ifndef VERSION_H
#define WASI_SDK_VERSION @WASI_SDK_VERSION@
#define WASI_LIBC_VERSION @WASI_LIBC_VERSION@
#endif
```
And then use `configure_file` : https://cmake.org/cmake/help/latest/command/configure_file.html

Then you would just need to set `WASI_LIBC_VERSION` and `WASI_SDK_VERSION` as cmake variables (presumably still by running version.py.. but without needing to add anything new to version.py I would hope).

Reply:
I like that better for the most part but I was worried about having a non-functional file cause problems. That approach means that we would add a non-functional `version.h` file to wasi-libc that kind of hangs around un-configured until it is seen by wasi-sdk. For those users who interact with wasi-libc directly (not through wasi-sdk), wouldn't they see an error if they included that un-configured `version.h`?

Reply:
Oh, sorry, the non-functional file is called `version.h.in`.. which is used to create the functional `version.h`, so they won't be confused.